app/service/rest.py

```diff
@@ -210,6 +217,72 @@ def get_service_notification_statistics(service_id):
     )
 
 
+@service_blueprint.route("/<uuid:service_id>/statistics/<string:start>/<int:days>")
+def get_service_notification_statistics_by_day(service_id, start, days):
+    return jsonify(
+        data=get_service_statistics_for_specific_days(service_id, start, int(days))
+    )
+
+
+def get_service_statistics_for_specific_days(service_id, start, days=1):
+    start_date = datetime.strptime(start, "%Y-%m-%d").date()
+
+    if days == 1:
+        stats = {}
+        stats[start] = statistics.format_statistics(
+            dao_fetch_stats_for_service_from_day(service_id, start_date)
+        )
+    else:
+        stats = {}
+        for d in range(days):
+            new_date = start_date - timedelta(days=d)
+            key = new_date.strftime("%Y-%m-%d")
+            stats[key] = statistics.format_statistics(
+                dao_fetch_stats_for_service_from_day(service_id, new_date)
+            )
```

Comment:
While I agree with Carlo on his point, there is also the issue of mixing front-end 'analytics' queries with back-end crud queries.  It goes beyond the scope of this story, but there is a risk that as the UI asks more and more fringy, exotic questions, that it starts to hog db time and back-end operations get impacted.  And this would be bad because back-end queues can fill up, and then we are actually losing data forever because it doesn't get written to the db etc.

We should ideally have a read replica, and the front end should be talking to that instead of the "real" db.   I guess such a change is out of scope until we are sending millions of messages a day, but all the same.

Reply:
I will review some ways to improve performance on the queries to help eliminate db server load.